src/math/exp_fcn.c

```diff
@@ -217,4 +221,52 @@ int32_t sofm_exp_int32(int32_t x)
 	}
 	return ts;
 }
+
+#define ONE_Q20         Q_CONVERT_FLOAT(1.0, 20)	  /* Use Q12.20 */
+#define TWO_Q27         Q_CONVERT_FLOAT(2.0, 27)	  /* Use Q5.27 */
+#define MINUS_TWO_Q27   Q_CONVERT_FLOAT(-2.0, 27)	  /* Use Q5.27 */
+
+/* Fixed point exponent function for approximate range -11.5 .. 7.6
+ * that corresponds to decibels range -100 .. +66 dB.
+ *
+ * The functions uses rule exp(x) = exp(x/2) * exp(x/2) to reduce
+ * the input argument for private small value exp() function that is
+ * accurate with input range -2.0 .. +2.0. The number of possible
+ * divisions by 2 is computed into variable n. The returned value is
+ * exp()^(2^n).
+ *
+ * Input  is Q5.27, -16.0 .. +16.0, but note the input range limitation
+ * Output is Q12.20, 0.0 .. +2048.0
+ */
+int32_t sofm_exp_fixed(int32_t x)
+{
+	int32_t xs;
+	int32_t y;
+	int32_t y0;
+	int i;
+	int n = 0;
+
+	if (x < Q_CONVERT_FLOAT(-11.5, 27))
```

Comment:
can we use a macro replace those fixed float convert? from my test, the division really costs many cycles.

Reply:
Yup, We can swap it out, saving about 93% of the cycles compared to Generic C. We can handle this in the future PR as the present one just addresses modifications for exponential. I hope you will consent to this.

Reply:
actually, I'd like use a simple macro like #define THERSHOULD Q_CONVERT_FLOAT(-11.5, 27) to replace the convert. because the result is a fixed statistic.